bundles/org.openhab.binding.fsinternetradio/src/main/java/org/openhab/binding/fsinternetradio/internal/FSInternetRadioDiscoveryParticipant.java

```diff
@@ -150,35 +158,41 @@ public DiscoveryResult createResult(RemoteDevice device) {
                 if (manufacturer != null) {
                     properties.put(PROPERTY_MANUFACTURER, manufacturer);
                 }
-                final String model = getModel(device);
+                final String model = getModel(device) != null ? getModel(device) : getFriendlyName(device);
```

Comment:
You should cache `getModel(device)` in a variable so you can reuse it.

Reply:
changed